code/modules/power/supermatter/supermatter_cascade_components.dm

```diff
@@ -0,0 +1,254 @@
+/turf/closed/indestructible/supermatter_wall
+	name = "wall"
+	desc = "Effectively impervious to conventional methods of destruction."
+	icon = 'icons/turf/walls.dmi'
+	icon_state = "crystal_cascade_1"
+	layer = AREA_LAYER
+	plane = ABOVE_LIGHTING_PLANE
+	opacity = FALSE
+	var/list/available_dirs = list(NORTH,SOUTH,EAST,WEST,UP,DOWN)
+	var/next_check = 0
+
+/turf/closed/indestructible/supermatter_wall/Initialize(mapload)
+	. = ..()
+	icon_state = "crystal_cascade_[rand(1,6)]"
+	START_PROCESSING(SSmachines, src)
+
+/turf/closed/indestructible/supermatter_wall/process()
+
+	if(next_check > world.time)
```

Comment:
use the cooldown defines here `COOLDOWN_DECLARE`  `COOLDOWN_START` `COOLDOWN_FINISHED`